### PR TITLE
move test to a dev-dependency and move test apis to transformer_test package

### PR DESCRIPTION
**Pull request by jakemac53**

### CODE REVIEW

File: test/assets_test.dart [hunk 1]
```diff
@@ -8,7 +8,7 @@ import 'dart:async';
 
 import 'package:barback/barback.dart';
 import 'package:code_transformers/assets.dart';
-import 'package:code_transformers/tests.dart';
+import 'package:transformer_test/transformer_test.dart';
```
Comment (sigmundch) [inline test/assets_test.dart]: many tools would consider transformer_test.dart as a test file. Maybe rename it to something else? Some ideas:
 * `transformer_testing/transformer_testing.dart`
 * `transformer_test/transformer_test_utils.dart`
 * `transformer_test/utils.dart`
 * `transformer_test/test_harness.dart`

**Comment by sigmundch** [comment]
other than the name suggestion, lgtm

File: test/assets_test.dart (hunk 1 above)
Comment (jakemac53) [inline test/assets_test.dart]: Ah crap thats a good point :(

Comment (jakemac53) [inline test/assets_test.dart]: I will just go with `utils` I think

**Pull request merged**